examples/Acoustics/compute_spectrum_production.py

```diff
@@ -0,0 +1,280 @@
+based on the PSU-WOPWOP 3.4.3 User Guide
+
+Input:
+The required input is set at the top of the main function in this script.
+Optional arguments can be modified directly in the acoustics_output function call.
+The only required input is the input file containing the pressure data at the observer location,
+typically of the form {caseID}_total_acoustics_v3.csv.
+
+Valid arguments to the optional inputs include:
+weighting: "A" None
+window: "Hann" None
+output: any string for the output file name
+observer_id: -1 corresponds to post-processing all observer ID's
+or an integer corresponding to a single observer ID present in the input file.
+
+Outputs:
+A csv file containing the OASPL for each observerID
+A plot of observerID vs OASPL
+PNG files with the spectral plots for each observerID
+"""
+
+import os
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+# Input corresponding environmental conditions
+GRID_UNIT = 1  # m
+SPEED_OF_SOUND = 340  # m/s, freestream
+RHO = 1.225  # kg/m^3, freestream
+REFERENCE_SOUND_PRESSURE = 20e-6  # Pa
+
+# Constants used for A-weighting
+K1 = 2.243e16
+K3 = 1.562
+F1 = 20.599
+F2 = 107.653
+F3 = 737.862
+F4 = 12194.220
+
+
+def read_csv_file(file_path):
+    """This function reads in the input file"""
+
+    df = pd.read_csv(file_path, skipinitialspace=True)
+    df = df.iloc[:, :-1]
+    data = df.to_dict(orient="list")
+    return data
+
+
+def apply_windowing_function(acoustics_data, i_observer, window):
+    """This function reads in the CSV input file and returns dict"""
+
+    pressure = np.asarray(acoustics_data[f"observer_{i_observer}_pressure"])
+    indices = np.nonzero(pressure)
+    non_zero_pressure = pressure[indices]
+    n = len(non_zero_pressure)  # number of time points in the time history
+
+    # window correction."None"= No window, "Hann"= Hann Window
+    if window == "Hann":
+        window_function = np.hanning(n)
+        fc = np.sqrt(n / sum(window_function**2))
+    else:
+        window_function = 1.0
+        fc = 1.0
+    # The windows that are actually applied to the time history are
+    # the window functions multiplied by their respective scaling factor
+    windowed_pressure = window_function * fc * non_zero_pressure
+    return windowed_pressure
+
+
+def compute_a_weight(f_m):
+    """This function computes the a_weight"""
+
+    w_cm = K1 * f_m**4 / ((f_m**2 + F1**2) ** 2 * (f_m**2 + F4**2) ** 2)
+    w_am = w_cm * K3 * f_m**4 / ((f_m**2 + F2**2) * (f_m**2 + F3**2))
+    return w_am
+
+
+def compute_spectrum(windowed_pressure, time_step_size, weighting):
+    """This function performs the FFT on the pressure signal and computes the OASPL and SPL spectrum"""
+
+    n = len(windowed_pressure)
+    physical_pressure = windowed_pressure * RHO * SPEED_OF_SOUND**2  # Pa
+    physical_time_step_size = time_step_size * GRID_UNIT / SPEED_OF_SOUND  # s
+    frequency_step_size = 1.0 / (n * physical_time_step_size)
```

Comment:
@thomascfd The following code seems to be the core technical content (ie, primary calculations from WOPWOP). Could you include some summary comments. Hoping this will help to point users in the right direction if they want/need to learn more.

Reply:
Added.